server/controllers/oauth.py

```diff
@@ -91,6 +112,13 @@ def authorize(*args, **kwargs):
     confirm = request.form.get('confirm', 'no')
     return confirm == 'yes'
 
+@oauth.route('/oauth/code')
+def oauth_code():
```

Comment:
This page doesn't access any user information (it just displays whatever is in the query params), so I don't see why logging in is required. You have to log in to get the code anyway.

The reason I made it a standard route is because of https://en.wikipedia.org/wiki/Post/Redirect/Get: If the user refreshes the code page nothing bad will happen, whereas if we return the response directly in the oob decorator they'll get a "confirm form resubmission" popup on refresh.